Structure_Engine/Modify/Flip.cs

```diff
@@ -47,10 +59,54 @@ public static Bar Flip(this Bar bar)
 
             Bar flipped = bar.ShallowClone();
 
+            // Flip Nodes
             Node tempNode = flipped.Start;
             flipped.Start = flipped.End;
             flipped.End = tempNode;
 
+            if (bar.IsVertical())
+                flipped.OrientationAngle = -bar.OrientationAngle + Math.PI;
+            else
+                flipped.OrientationAngle = -bar.OrientationAngle;
+
+            // Flip releases
+            BarRelease flippedRelease = bar.Release?.ShallowClone();
+            if (flippedRelease != null)
+            {
+                Constraint6DOF tempRelease = flippedRelease.StartRelease;
+                flippedRelease.StartRelease = flippedRelease.EndRelease;
+                flippedRelease.EndRelease = tempRelease;
+                flipped.Release = flippedRelease;
+            }
+
+            // Flip section property
+            if (bar.SectionProperty != null)
+            {
+                ISectionProperty flippedSectionProperty = Flip(bar.SectionProperty);
+                flipped.SectionProperty = flippedSectionProperty;
+            }
+
+            // Flip offsets
+            if (bar.Offset != null)
+            {
+                Offset flippedOffset = new Offset()
```

Comment:
Flipped offset is never assigned! 
```suggestion
                flipped.Offset = new Offset()
```